napari/layers/shapes/shapes.py

```diff
@@ -559,20 +586,35 @@ def mode(self, mode):
                 'hold <space> to pan/zoom, '
                 f'press <{BACKSPACE}> to remove selected'
             )
+            self.mouse_drag_callbacks.append(select)
+            self.mouse_move_callbacks.append(highlight)
         elif mode in [Mode.VERTEX_INSERT, Mode.VERTEX_REMOVE]:
             self.cursor = 'cross'
             self.interactive = False
             self.help = 'hold <space> to pan/zoom'
+            if mode == Mode.VERTEX_INSERT:
+                self.mouse_drag_callbacks.append(vertex_insert)
+            else:
+                self.mouse_drag_callbacks.append(vertex_remove)
+            self.mouse_move_callbacks.append(highlight)
         elif mode in [Mode.ADD_RECTANGLE, Mode.ADD_ELLIPSE, Mode.ADD_LINE]:
             self.cursor = 'cross'
             self.interactive = False
             self.help = 'hold <space> to pan/zoom'
+            if mode == Mode.ADD_RECTANGLE:
```

Comment:
For this list I would consider creating a mode2function dictionary and having a single call to append

Reply:
this will get redone in #1048 so i'd rather leave it for now